include/jsoncons/json.hpp

```diff
@@ -654,21 +658,88 @@ class basic_json
         };
 
     private:
-        static const size_t data_size = static_max<sizeof(uinteger_data),sizeof(double_data),sizeof(small_string_data), sizeof(string_data), sizeof(array_data), sizeof(object_data)>::value;
-        static const size_t data_align = static_max<JSONCONS_ALIGNOF(uinteger_data),JSONCONS_ALIGNOF(double_data),JSONCONS_ALIGNOF(small_string_data),JSONCONS_ALIGNOF(string_data),JSONCONS_ALIGNOF(array_data),JSONCONS_ALIGNOF(object_data)>::value;
 
-        typedef typename std::aligned_storage<data_size,data_align>::type data_t;
+        union
+        {
+            base_data base_data_;
+            null_data null_data_;
+            bool_data bool_data_;
+            integer_data integer_data_;
+            uinteger_data uinteger_data_;
+            double_data double_data_;
+            small_string_data small_string_data_;
+            string_data string_data_;
+            byte_string_data byte_string_data_;
+            array_data array_data_;
+            object_data object_data_;
+            empty_object_data empty_object_data_;
+        };
```

Comment:
I removed the aligned storage and replaced it with union members of all the different variant types. If this project was targeted for c++17, we could use std::variant, but this should do for c++11.

A programmer can now inspect the variant data conveniently in debugger.